Source/Core/DiscIO/Volume.h

```diff
@@ -43,31 +84,34 @@ class IVolume
 	virtual bool CheckIntegrity() const { return false; }
 	virtual bool ChangePartition(u64 offset) { return false; }
 
-	// Increment CACHE_REVISION if the code below is modified (ISOFile.cpp & GameFile.cpp)
-	enum ECountry
-	{
-		COUNTRY_EUROPE = 0,
-		COUNTRY_JAPAN,
-		COUNTRY_USA,
-		COUNTRY_AUSTRALIA,
-		COUNTRY_FRANCE,
-		COUNTRY_GERMANY,
-		COUNTRY_ITALY,
-		COUNTRY_KOREA,
-		COUNTRY_NETHERLANDS,
-		COUNTRY_RUSSIA,
-		COUNTRY_SPAIN,
-		COUNTRY_TAIWAN,
-		COUNTRY_WORLD,
-		COUNTRY_UNKNOWN,
-		NUMBER_OF_COUNTRIES
-	};
-
 	virtual ECountry GetCountry() const = 0;
 	virtual u64 GetSize() const = 0;
 
 	// Size on disc (compressed size)
 	virtual u64 GetRawSize() const = 0;
+
+protected:
+	template <u32 N> std::string DecodeString(const char(&data)[N]) const
+	{
+		// strnlen to trim NULLs
+		std::string string(data, strnlen(data, sizeof(data)));
```

Comment:
The files that are decoded using this function usually have NULs at the end of strings, but it's not guaranteed.